mavisp/modules.py

```diff
@@ -105,76 +105,68 @@ def ingest(self, mutations):
-                raise MAVISpMultipleError(warning=warnings,
-                                        critical=[MAVISpCriticalError(this_error)])
-            all_methods.extend(method_dirs)
-
-        if len(all_methods) != len(set(all_methods)):
-                this_error = f"Only using one single instance of any given method is supported"
-                raise MAVISpMultipleError(warning=warnings,
-                                        critical=[MAVISpCriticalError(this_error)])
-
-        for model in models:
-
-            method_dirs = os.listdir(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model))
-
-            for method_dir in method_dirs:
-                self.methods[method_dir].parse(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model, method_dir))
-                self.data = self.data.join(self.methods[method_dir].data)
-
-        keys = [ k for k in self.data.keys() if k.startswith('Stability') ]
-
-        # check if we have FoldX column
-        foldx_col = [k for k in keys if 'FoldX' in k]
-        if len(foldx_col) == 1:
-            foldx_header = foldx_col[0]
-        else:
-            foldx_header = None
-
-        if any(['Rosetta' in k for k in keys]):
-            rosetta_col = [k for k in keys if 'Rosetta' in k]
-            assert len(rosetta_col) == 1
-            rosetta_header = rosetta_col[0]
-        else:
-            rosetta_header = None
+            for model in models:
+                method_dirs = os.listdir(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model))
+                if not set(method_dirs).issubset(set(self.methods.keys())):
+                    this_error = f"One or more {self.name} methods are not supported"
+                    raise MAVISpMultipleError(warning=warnings,
+                                            critical=[MAVISpCriticalError(this_error)])
+                all_methods.extend(method_dirs)
+
+                if len(all_methods) != len(set(all_methods)):
+                                this_error = f"Only using one single instance of any given method is supported"
+                                raise MAVISpMultipleError(warning=warnings,
+                                                        critical=[MAVISpCriticalError(this_error)])
+            for model in models:
+
+                method_dirs = os.listdir(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model))
+                for method_dir in method_dirs:
+                    parsed_data, this_warnings = self.methods[method_dir].parse(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model, method_dir))
+                    warnings += this_warnings
+                    self.data = self.data.join(parsed_data,rsuffix = f" using {method} method")
```

Comment:
I think this logic risks being a bit fragile, because at each iteration we add to the same data frame altogether (`self.data`). For instance, this means that when we parse data of the second model we will already have one FoldX Stability and one Rosetta Stability column. This makes lines 143-158 a bit unpredictable; are we sure the last column we want to use will be in all cases the last one available, especially if we have cases in which we are missing Rosetta, FoldX or RaSP for some cases?

I think we should change the logic in this way:

```
for every model:
    create a new dataframe with `mutations` as index (we can call it e.g. `model_data`)
    merge in FoldX/Rosetta/RaSP data if available as you do now
    identify columns we need (in this case you should be able to reuse my old code to do so. My old code had na `assert` on the number of columns because it should always be one)    
    generate classification column in the `model_data` dataframe. This means we now have a data frame with just Rosetta, FoldX and classification columns for this specific model
    join this data frame in self.data
```